yt/python/yt/wrapper/query_commands.py

```diff
@@ -195,3 +204,303 @@ def format_time(time):
         params=params,
         format=format,
         client=client)
+
+# Helpers
+
+
+def get_query_url(query_id, client=None):
+    proxy_url = get_proxy_address_url(required=False, client=client)
+    if not proxy_url:
+        return None
+
+    proxy, cluster_path = get_proxy_and_cluster_path(client=client)
+
+    return get_config(client)["proxy"]["query_link_pattern"].format(
+        proxy=proxy,
+        cluster_path=cluster_path,
+        id=query_id)
+
+
+class QueryState(object):
+    """State of the query (simple wrapper for the string name)."""
+    def __init__(self, name):
+        self.name = name
+
+    def is_finished(self):
+        return self.name in ("aborted", "completed", "failed", "draft")
+
+    def is_unsuccessfully_finished(self):
+        return self.name in ("aborted", "failed")
```

Comment:
I would make inverted method `is_successfully_finished`.

Current approach suggests that if you want to check success, you call `if not state.is_unsuccessfully_finished` and I find this ugly.

Reply:
No problem.
Note that there is the similar method in OperationState, you might want to refactor it too for similarity.